frontend/public/components/secrets/create-secret.tsx

```diff
@@ -293,24 +251,62 @@ class SSHAuthSubform extends React.Component<SSHAuthSubformProps, SSHAuthSubform
     }, () => this.props.onChange(this.state));
   }
   render() {
-    return <div className="form-group">
-      <label className="control-label" htmlFor="ssh-privatekey">SSH Private Key</label>
-      <div>
-        <FileInput onChange={this.onFileChange} />
-        <p className="help-block">Upload your private SSH key file.</p>
-        <textarea className="form-control co-create-secret-form__textarea"
-          id="ssh-privatekey"
-          name="privateKey"
-          onChange={this.changeData}
-          value={this.state['ssh-privatekey']}
-          aria-describedby="ssh-privatekey-help"
-          required />
-        <p className="help-block" id="ssh-privatekey-help">Private SSH key file for Git authentication.</p>
-      </div>
-    </div>;
+    return <DropableFileInput
+      onChange={this.onFileChange}
+      inputFileData={this.state['ssh-privatekey']}
+      id="ssh-privatekey"
+      label="SSH Private Key"
+      inputFieldHelpText="Drag and drop your private SSH key here or browse to upload it."
+      textareaFieldHelpText="Private SSH key file for Git authentication." />;
+  }
+});
+
+class SourceSecretForm extends React.Component<SourceSecretFormProps, SourceSecretFormState> {
```

Comment:
Do you mind moving this back where it was to avoid the large diff? (Assuming you didn't move it here for some other reason.)